murfi-rt-PyProject/scripts/feedback.sh

```diff
@@ -126,29 +79,39 @@ then
+    #yes n | gzip $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.nii
     
-    #../scripts/img/img-00001-00211.nii ../scripts/img/img-00001-00212.nii ../scripts/img/img-00001-00213.nii ../scripts/img/img-00001-00214.nii ../scripts/img/img-00001-00215.nii ../scripts/img/img-00001-00216.nii ../scripts/img/img-00001-00217.nii ../scripts/img/img-00001-00218.nii ../scripts/img/img-00001-00219.nii ../scripts/img/img-00001-00220.nii ../scripts/img/img-00001-00221.nii ../scripts/img/img-00001-00222.nii ../scripts/img/img-00001-00223.nii ../scripts/img/img-00001-00224.nii ../scripts/img/img-00001-00225.nii ../scripts/img/img-00001-00226.nii ../scripts/img/img-00001-00227.nii ../scripts/img/img-00001-00228.nii ../scripts/img/img-00001-00229.nii ../scripts/img/img-00001-00230.nii ../scripts/img/img-00001-00231.nii ../scripts/img/img-00001-00232.nii ../scripts/img/img-00001-00233.nii ../scripts/img/img-00001-00234.nii ../scripts/img/img-00001-00235.nii ../scripts/img/img-00001-00236.nii ../scripts/img/img-00001-00237.nii ../scripts/img/img-00001-00238.nii ../scripts/img/img-00001-00239.nii ../scripts/img/img-00001-00240.nii ../scripts/img/img-00001-00241.nii ../scripts/img/img-00001-00242.nii ../scripts/img/img-00001-00243.nii ../scripts/img/img-00001-00244.nii ../scripts/img/img-00001-00245.nii ../scripts/img/img-00001-00246.nii ../scripts/img/img-00001-00247.nii ../scripts/img/img-00001-00248.nii ../scripts/img/img-00001-00249.nii 1.2
+    # get all volumes of resting data (no matter how many) merged into 1 .nii.gz file
+    # NOTE: the -00002 extensipn will likely need to be adjusted depending on where this scan falls in the protocol
+    fslmerge -tr $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.nii.gz $subj_dir/img/img-00002* 1.2
+    chmod 777 $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.nii.gz 
+
+    # figure out how many volumes of resting state data there were to be used in ICA
+    restvolumes=$(fslnvols $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.nii.gz)
 
     echo "+ computing resting state networks this will take about 25 minutes"
     echo "+ started at: $(date)"
     
+    # update FEAT template with paths and # of volumes of resting state run
     cp $fsl_scripts/rest_template.fsf $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.fsf
     DATA_path=$subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.nii.gz
     OUTPUT_dir=$subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'
-    sed -i "s#DATA#$DATA_path#g" $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.fsf
+    sed -i "s#DATA#$subj_dir_absolute/rest/${subj}_${ses}_task-rest_${run}_bold.nii.gz#g" $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.fsf
     sed -i "s#OUTPUT#$OUTPUT_dir#g" $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.fsf
+
+    # update fsf to match number of rest volumes
+    sed -i "s/set fmri(npts) 248/set fmri(npts) ${restvolumes}/g" $subj_dir/rest/$subj'_'$ses'_task-rest_'$run'_bold'.fsf
```

Comment:
Tried to make this a little more flexible, so now it will update .fsf to run ICA on whatever number of BOLD volumes MURFI can find for the given run. Is this okay? Or should we throw some warnings/errors if this isn't the expected number of volumes for the complete resting state run?

Reply:
It would be nice to through some warnings and then continue with this so we can make a note

Reply:
great, will add those!

Reply:
added based on expectation of 250 volumes